### PR TITLE
Use latest "message" api in vest

### PR DESCRIPTION
**Pull request by ealush**
Recently, a contributor to Vest added a new implementation to enforce, supporting providing a message (https://github.com/ealush/vest/pull/929).

This can be used to make the custom message flow more declarative, and instead of throwing a message on a condition, we can simply use enforce.

I added a few changes that reflect how I usually use Vest that might reduce the amount of code. Feel free to ignore these changes.

*Note* - I wasn't sure about simulating the data and actually testing the validation logic, I may have missed there - but in terms of Vest's capabilities, everything there should be possible.

### CODE REVIEW

File: src/app/validators/company.async-validations.ts [hunk 1]
```diff
@@ -1,53 +1,78 @@
...
-      // No fvs? This is almost certainly a bug. Remember to provide a context.
-      : (fein: string) => Promise.resolve<FeinValidationResponse>({ fein });
+  const fvs = (<AppValidationContext>context)?.feinValidationService;
+  /** FEIN check function (or FN returning "not found" result) */
+  const checkFein = fvs
+    ? fvs.check.bind(fvs)
+    : // No fvs? This is almost certainly a bug. Remember to provide a context.
+      (fein: string) => Promise.resolve<FeinValidationResponse>({ fein });
 
+  omitWhen(!isGoodFein(model.fein), () => {
```
Comment (ealush) [inline src/app/validators/company.async-validations.ts:52]: omitWhen allows these tests to be skipped completely, without marking the suite as invalid. This means we don't need the condition inside each of the tests.
https://vestjs.dev/docs/writing_your_suite/including_and_excluding/omitWhen

File: src/app/validators/company.async-validations.ts [hunk 2]
```diff
@@ -1,53 +1,78 @@
...
-          throw `Does not match "${name}", the name registered with the IRS for this tax number`;
-        }
+
+        enforce(name)
+          .message(
+            `Does not match "${name}", the name registered with the IRS for this tax number`
+          )
+          .isNotBlank()
+          .notEquals('*')
+          .notEquals(model.legalName.toUpperCase());
```
Comment (ealush) [inline src/app/validators/company.async-validations.ts:67]: This is pretty much the same, only declarative. I think we could even do:

```js
        enforce(name)
          .message(
            `Does not match "${name}", the name registered with the IRS for this tax number`
          )
          .isNotBlank()
          .notInside(['*', model.legalName.toUpperCase()]);
```

File: src/app/validators/company.async-validations.ts [hunk 3]
```diff
@@ -1,53 +1,78 @@
...
     });
 
-
     test('fein', 'Tax Number is not registered with the IRS', async () => {
-      if (isGoodFein(model.fein)) {
-        const response = await checkFein(model.fein!);
-        // console.log('fein checkFein response', response);
-        enforce(response.feinName).isNotNullish();
-      }
-      return ''; // All other cases are either OK or the test does not apply
```
Comment (ealush) [inline src/app/validators/company.async-validations.ts:51]: This is not needed. Vest really ignores any return value other than:
- thrown Error
- thrown String
- `false` (literal false, not falsy values)
- Promise rejection

https://vestjs.dev/docs/writing_tests/using_the_test_function#how-to-fail-a-test

**Comment by wardbell** [comment]
Thanks a ton!

**Pull request merged**